livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -407,6 +408,13 @@ private String callRestAPI(String targetURL, String method, String jsonData)
       response = new ResponseEntity(e.getResponseBodyAsString(), e.getStatusCode());
       LOGGER.error(String.format("Error with %s StatusCode: %s",
           response.getStatusCode().value(), e.getResponseBodyAsString()));
+    } catch (RestClientException e) {
+      // Exception happens when kerberos is enabled.
+      if (e.getMessage().contains("404 Not Found")) {
+        throw new SessionNotFoundException("Session not found when kerberos is enabled ");
```

Comment:
Great point, looks like the useful message is hidden in `HttpClientErrorException.getResponseBodyAsString()`, update the PR